### PR TITLE
Ensure vulkan subgroups are disabled for MoltenVK

### PR DESCRIPTION
**Pull request by RevoluPowered**
We found they don't work on intel macbooks properly at all.

Possible future solutions:
- update to moltenvk.
- update to spirv may resolve it.

### CODE REVIEW

File: servers/rendering/renderer_rd/shaders/forward_clustered/scene_forward_clustered.glsl [hunk 1]
```diff
@@ -1775,15 +1773,15 @@ void fragment_shader(in SceneData scene_data) {
...
 			} else {
-				shadow = float(shadow1 >> ((i - 4) * 8) & 0xFF) / 255.0;
+				shadow = float(shadow1 >> ((i - 4u) * 8u) & 0xFFu) / 255.0;
 			}
 
 			shadow = shadow * directional_lights.data[i].shadow_opacity + 1.0 - directional_lights.data[i].shadow_opacity;
 #endif
 
-			blur_shadow(shadow);
+			//blur_shadow(shadow);
```
Comment (clayjohn) [inline servers/rendering/renderer_rd/shaders/forward_clustered/scene_forward_clustered.glsl:1784]: You can leave this in, we are still figuring out if we will support shadow blurring this way or not

Comment (RevoluPowered) [reply servers/rendering/renderer_rd/shaders/forward_clustered/scene_forward_clustered.glsl]: Fixed, that should not be in there!

**Review by clayjohn — APPROVED** [review]
Looks good to me. In my mind this is more of stopgap until we can figure out why exactly subgroups are crashing on MacOS. But we should fix the crash now and then fix subgroups if/when we can

**Comment by RandomShaper** [comment]
Yeah, let's keep an eye on this to be sure we do the right thing. Aside, the changes about unsigned correctness would belong in a separate commit/PR, but I wouldn't block merging on that. Just for next time.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**